### PR TITLE
Check attributes fix

### PR DESCRIPTION
**Pull request by sufka861**
fixed checkAttribute
- used a library called lodash _.isEqual to compare to objects of customAttributes
- kept the same logic, but added the option that there is customAttributes being passed in the endUserReq
if the customAttribute is empty/null then it will compare only the defaults

### CODE REVIEW

File: Service/utils.js [hunk 1]
```diff
@@ -40,16 +41,21 @@ const checkAttributes = (endUserReq, experiment, next) => {
...
+                browser === experiment.testAttributes.browser.value &&
+                device === experiment.testAttributes.device.value;
+            let attributes = {"location": geo.country, "browser": browser, "device": device};
+            let customAttResult;
+            if (customAttributes) {
+                customAttResult = _.isEqual(customAttributes, experiment.customAttributes);
+                attributes = {...attributes, ...customAttributes};
+            }
             const attReqCountResult = incAttributeReqCount(experiment.experimentId, attributes);
-            // if (!attReqCountResult) throw new ServerUnableError("attReqCountResult");
```
Comment (AtomicManic) [inline Service/utils.js:51]: why commented? looks like you need this

File: Service/utils.js [hunk 2]
```diff
@@ -40,16 +41,21 @@ const checkAttributes = (endUserReq, experiment, next) => {
     try {
         const geo = getLocation(getClientIP(endUserReq));
         const {browser, device} = getBrowserDevice(endUserReq);
+        const customAttributes = endUserReq.customAttributes;
         if (geo && browser && device) {
-            const result =
-                geo.country === experiment.testAttributes.location[0] &&
-                browser === experiment.testAttributes.browser[0] &&
-                device === experiment.testAttributes.device[0];
-            const customAttributes = endUserReq.customAttributes;
```
Comment (AtomicManic) [inline Service/utils.js:48]: what if we don't have custom attributes?

**Review by AtomicManic — COMMENTED** [review]
Looking good overall. Wrote some comments, LMK when you made the changes =]

File: Service/utils.js (hunk 2 above)
Comment (sufka861) [reply Service/utils.js:48]: then it should be null or empty

File: Service/utils.js (hunk 1 above)
Comment (sufka861) [reply Service/utils.js:51]: it was allready removed

**Review by AtomicManic — APPROVED** [review]

**Pull request merged**